Objects/object.c

```diff
@@ -1715,10 +1715,16 @@ _PyObject_GenericGetAttrWithDict(PyObject *obj, PyObject *name,
             dict = (PyObject *)_PyObject_GetManagedDict(obj);
         }
         else {
+            Py_BEGIN_CRITICAL_SECTION(obj);
```

Comment:
In general, we don't need to lock something if we're going to use one of the `_Py_atomic` APIs, and vice versa.